OISF/suricata
# Pcre2 3194 v18

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/3194
https://redmine.openinfosecfoundation.org/issues/4446

Describe changes:
- migrate all pcre1 code to pcre2
- fuzz: builds fuzz_sigpcap_aware only when asked (ie ./configure --enable-fuzztargets)
- qa : remove some obsolete files
- pcrexform : checks that we have exactly one substring capture in a signature
- transforms : use generic `Free` functions everywhere

suricata-verify-pr: 502

https://github.com/OISF/suricata-verify/pull/502

Replaces https://github.com/OISF/suricata/pull/6330 with code comments taken into account

Should I make a PR with the first 4 preliminary commits ? (like transform: use generic Free function)
The completeness of the commit about `PCRE2_ERROR_UNSET` needs to be carefully reviewed : should it become the generic case ?

I bet CI commit-check will fail
